dbms/src/Flash/Mpp/LocalRequestHandler.h

```diff
@@ -25,22 +25,24 @@ struct LocalRequestHandler
         std::function<void(bool, const String &)> && notify_write_done_,
         std::function<void()> && notify_close_,
         std::function<void()> && add_local_conn_num_,
-        ReceiverChannelWriter && channel_writer_)
+        const std::string & req_info_,
+        ReceivedMessageQueue * msg_queue_)
         : notify_write_done(std::move(notify_write_done_))
         , notify_close(std::move(notify_close_))
         , add_local_conn_num(std::move(add_local_conn_num_))
-        , channel_writer(std::move(channel_writer_))
+        , req_info(req_info_)
+        , msg_queue(msg_queue_)
     {}
 
     template <bool is_force>
     bool write(size_t source_index, const TrackedMppDataPacketPtr & tracked_packet)
     {
-        return channel_writer.write<is_force>(source_index, tracked_packet);
+        return msg_queue->pushFromLocal<is_force>(source_index, req_info, tracked_packet, ReceiverMode::Local);
```

Comment:
Since the function name is `pushFromLocal`, where is no need to pass `ReceiverMode::Local`

Reply:
The `readLoop` and `AsyncRequestHandlerv1` still use this interface.